docs/guides/unity-catalog-azure.md

```diff
@@ -0,0 +1,422 @@
+
+Databricks Unity Catalog brings fine-grained governance and security to Lakehouse data using a familiar, open interface. You can use Terraform to deploy the underlying cloud resources and Unity Catalog objects automatically, using a programmatic approach.
+
+This guide uses the following variables in configurations:
+
+- `databricks_workspace_url`: Value of `workspace_url` attribute from [databricks_mws_workspaces](../resources/mws_workspaces.md#attribute-reference) resource.
+- `tenant_id`: The Azure AD tenant ID should be used for the [azuread](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs) provider.
+- `subscription_id`: The Azure subscription ID should be used for the [azurerm](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs) provider
+
+This guide is provided as-is and you can use this guide as the basis for your custom Terraform module.
+
+To get started with Unity Catalog, this guide takes you throw the following high-level steps:
+- [Deploying pre-requisite resources and enabling Unity Catalog (Azure Preview)](#deploying-pre-requisite-resources-and-enabling-unity-catalog-azure-preview)
+  - [Provider initialization](#provider-initialization)
+  - [Configure Azure objects](#configure-azure-objects)
+  - [Create a Unity Catalog metastore and link it to workspaces](#create-a-unity-catalog-metastore-and-link-it-to-workspaces)
+  - [Create Unity Catalog objects in the metastore](#create-unity-catalog-objects-in-the-metastore)
+  - [Configure external tables and credentials](#configure-external-tables-and-credentials)
+  - [Configure Unity Catalog clusters](#configure-unity-catalog-clusters)
+
+## Provider initialization
+
+Initialize the 3 providers to set up the required resources. See [Databricks provider authentication](../index.md#authenticating-with-hostname,-username,-and-password), [Azure AD provider authentication](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs#authenticating-to-azure-active-directory) and [Azure provider authentication](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#authenticating-to-azure) for more details.
+
+```hcl
+terraform {
+  required_providers {
+    databricks = {
+      source = "databrickslabs/databricks"
+    }
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>2.92.0"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "~>2.15.0"
+    }
+  }
+}
+
+provider "azuread" {
+  tenant_id = var.tenant_id
+}
+
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  features {}
+}
+
+// initialize provider at workspace level, to create UC resources
+provider "databricks" {
+  alias    = "workspace"
+  host     = var.databricks_workspace_url
+}
+```
+
+Define the required variables
+
+```hcl
+variable "databricks_workspace_url" {}
+
+variable "tags" {
+  default = {}
+}
+
+variable "reuse_rg" {
```

Comment:
remove this property for this guide, so that it's simpler. also, why don't you do this:

```hcl
variable "databricks_resource_id" {
  description = "The Azure resource ID for the databricks workspace deployment."
}

locals {
  resource_regex            = "(?i)subscriptions/.+/resourceGroups/(.+)/providers/Microsoft.Databricks/workspaces/(.+)"
  resource_group            = regex(local.resource_regex, var.databricks_resource_id)[0]
  databricks_workspace_name = regex(local.resource_regex, var.databricks_resource_id)[1]
}

data "azurerm_resource_group" "this" {
  name = local.resource_group
}

# Most likely, dependent resources would have
# ...
    name = replace(data.azurerm_resource_group.this.name, "rg", "akv")
    resource_group_name = data.azurerm_resource_group.this.name
    location = data.azurerm_resource_group.this.location
    tags = data.azurerm_resource_group.this.tags
# ...
```

Reply:
done, down to 1 variable now